backend/app/api/v1/endpoints/cpt/cptJobs.py

```diff
@@ -25,161 +26,191 @@
-    """
-    return (offset + requested_size) <= total_count
+    if start_date > end_date:
+        return Response(
+            content=json.dumps({"error": "start_date must be before end_date"}),
+            status_code=422,
+        )
 
+    filter_dict = get_dict_from_qs(filter) if filter else {}
+    prod_list = filter_dict.pop("product", list(productsFilter.keys()))
+
+    results = await asyncio.gather(
+        *[
+            fetch_data_limited(
+                product, start_date, end_date, filter=filter, is_filter=True
+            )
+            for product in prod_list
+        ]
+    )
+
+    total_dict, summary_dict, result_dict = (
+        {},
+        {"success": 0, "failure": 0, "other": 0, "total": 0},
+        {},
+    )
+
+    for result in results:
+        total_dict[result.get("product", "")] = result.get("total", 0)
+
+        for key, value in result.get("summary", {}).items():
+            summary_dict[key] += value
+
+        for item in result.get("data", []):
+            key, values = item["key"], item["value"]
+            # If the key already exists, merge the values
+            if key in result_dict:
+                if isinstance(values[0], str):
+                    existing_values = {v.lower(): v for v in result_dict[key]}
+                    for val in values:
+                        if val.lower() not in existing_values and val != "":
+                            result_dict[key].append(val)
+                else:
+                    # For numbers (version), just avoid duplicates
+                    result_dict[key] = list(set(result_dict[key] + values))
+            else:
+                result_dict[key] = [s for s in values if str(s).strip()]
+
+    merged_result = [
+        {"key": k, "value": v, "name": FILEDS_DISPLAY_NAMES.get(k, k)}
+        for k, v in result_dict.items()
+    ]
 
-def calculate_remaining_data(total_count, offset, requested_size):
-    """
-    Calculate the remaining number of data items that can be fetched based on the requested size.
+    response = {
+        "startDate": str(start_date),
+        "endDate": str(end_date),
+        "filterData": merged_result,
+        "summary": summary_dict,
+        "total": sum(int(v) for v in total_dict.values()),
+    }
 
-    Args:
-        total_count (int): Total number of available records.
-        offset (int): The starting position in the dataset.
-        requested_size (int): The number of records requested.
+    # if pretty:
+    #     json_str = json.dumps(response, indent=4)
+    #     return Response(content=json_str, media_type="application/json")
 
-    Returns:
-        int: The number of records that can be fetched, which may be less than or equal to requested_size.
-    """
-    available_data = total_count - offset  # Data available from the offset
-    return min(available_data, requested_size)
+    # jsonstring = json.dumps(response)
+    # return jsonstring
```

Comment:
Remove or re-instate the commented code. I think this backend "pretty" query parameter is rather silly given that presentation should be a client function, and the `ORJSONResponse` seems cleaner than returning a dumped (serialized) JSON string, but FastAPI can handle just returning a `dict` to represent a JSON response without any help, and that seems like the simplest and most consistent implementation.